crates/router/src/core/payments.rs

```diff
@@ -3324,13 +3324,19 @@ pub async fn get_aggregates_for_payments(
     time_range: common_utils::types::TimeRange,
 ) -> RouterResponse<api::PaymentsAggregateResponse> {
     let db = state.store.as_ref();
+
     let intent_status_with_count = db
         .get_intent_status_with_count(merchant.get_id(), profile_id_list, &time_range)
         .await
-        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+        .map_err(|err| {
+          
+            errors::ApiErrorResponse::InternalServerError
+                .attach_printable(format!("Error fetching payment aggregates: {:?}", err))
+        })?;
```

Comment:
Can we directly do something like:
```
.change_context(errors::ApiErrorResponse::InternalServerError)
.attach_printable("message")?;
```
Please check

Reply:
Yes, that’s a great suggestion! Using .change_context(errors::ApiErrorResponse::InternalServerError) along with .attach_printable("message")?; will provide a clearer context for any internal errors that occur while fetching the payment intent status.

This approach enhances error handling and improves debuggability by providing a more informative message. I’ll implement this change in the get_aggregates_for_payments function. Thank you for the recommendation!